### PR TITLE
filter cx responses by date

### PR DESCRIPTION
**Pull request by ryanwoldatwork**

### CODE REVIEW

File: app/controllers/api/v1/cx_responses_controller.rb [hunk 1]
```diff
@@ -9,13 +9,16 @@ def index
         page_size = 500 if page_size <= 0 || page_size > 5000
 
         begin
-          start_date = params[:start_date] ? Date.parse(params[:start_date]).to_date : Date.parse("2023-10-01").to_date
+          start_date = params[:start_date] ? Date.parse(params[:start_date]).to_date : Date.parse("2024-10-01")
```
Comment (ryanwoldatwork) [inline app/controllers/api/v1/cx_responses_controller.rb:12]: already a date

File: app/controllers/api/v1/cx_responses_controller.rb [hunk 2]
```diff
@@ -9,13 +9,16 @@ def index
...
           end_date = params[:end_date] ? Date.parse(params[:end_date]).to_date : 1.day.from_now
         rescue StandardError
           render json: { error: { message: "invalid date format, should be 'YYYY-MM-DD'", status: 400 } }, status: :bad_request and return
         end
 
-        cx_responses = CxResponse.page(page_number).per(page_size)
+        cx_responses = CxResponse
+          .where(created_at: start_date.beginning_of_day..end_date.end_of_day)
```
Comment (ryanwoldatwork) [inline app/controllers/api/v1/cx_responses_controller.rb:19]: apply the date filter

File: app/models/cx_collection_detail.rb [hunk 3]
```diff
@@ -4,7 +4,7 @@ class CxCollectionDetail < ApplicationRecord
   belongs_to :cx_collection
   belongs_to :service_stage, optional: true
   has_many :cx_responses, dependent: :delete_all
-  has_one :cx_collection_detail
+  has_many :cx_collection_detail_uploads
```
Comment (ryanwoldatwork) [inline app/models/cx_collection_detail.rb:7]: correct this.  
it doesn't have one of itself

File: spec/factories/cx_collection_detail.rb [hunk 4]
```diff
@@ -6,7 +6,7 @@
 
     trait :with_cx_collection_detail_upload do
       after(:create) do |cx_collection_detail|
-        FactoryBot.create(:cx_collection_detail_upload, cx_collection_detail: cx_collection_detail)
```
Comment (ryanwoldatwork) [inline spec/factories/cx_collection_detail.rb:9]: given the key matches, it doesn't need to passed in.

this is optional, and primarily aesthetic

**Review by sanason — APPROVED** [review]
👍

**Pull request merged**